control-plane-operator/controllers/hostedcontrolplane/pki/oauth.go

```diff
@@ -1,11 +1,19 @@
 package pki
 
 import (
+	"net"
+
 	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/config"
 	corev1 "k8s.io/api/core/v1"
 )
 
 func (p *PKIParams) ReconcileOAuthServerCert(secret, ca *corev1.Secret, ownerRef config.OwnerRef, externalOAuthAddress string) error {
-	oauthHostNames := []string{externalOAuthAddress}
-	return reconcileSignedCertWithAddresses(secret, ca, ownerRef, "openshift-oauth", []string{"openshift"}, X509DefaultUsage, X509UsageClientServerAuth, oauthHostNames, nil)
+	var dnsNames, ips []string
+	oauthIP := net.ParseIP(externalOAuthAddress)
```

Comment:
Could we add  a unittest that verifies this correctly generates certs for both ips and dns names?